include/ensmallen_bits/problems/zdt3_function.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file zdt3_function.hpp
+ * @author Nanubala Gnana Sai
+ *
+ * Implementation of the third ZDT(Zitzler, Deb and Thiele) test.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_PROBLEMS_ZDT_THREE_FUNCTION_HPP
+#define ENSMALLEN_PROBLEMS_ZDT_THREE_FUNCTION_HPP
+
+namespace ens {
+namespace test {
+
+/**
+ * The ZDT3 function, defined by:
+ * \f[
+ * g(x) = 1 + 9(\sum_{i=2}^{n} x_i )/(n-1)
+ * f_1(x) = x_i
+ * h(f_1,g) = 1 - \sqrt{f_1/g}  - (f_1/g)sin(10\pi f_1)
+ * f_2(x) = g(x) * h(f_1, g)
+ * \f]
+ *
+ * This is a 30-variable problem(n = 30) with a number
+ * of disconnected optimal fronts.
+ *
+ * Bounds of the variable space is:
+ * 0 <= x_i <= 1 for i = 1,...,n.
+ *
+ * This should be optimized to g(x) = 1.0, at:
+ *
+ * x_1* in [0.0000, 0.0830] OR
+ * x_1* in [0.1822, 0.2577] OR
+ * x_1* in [0.4093, 0.4538] OR
+ * x_1* in [0.6183, 0.6525] OR
+ * x_1* in [0.8233, 0.8518].
+ *
+ * x_i* = 0 for i = 2,...,n.
+ *
+ * For more information, please refer to:
+ *
+ * @code
+ * @article{Zitzler2000,
+ *   title   = {Comparison of multiobjective evolutionary algorithms:
+ *              Empirical results},
+ *   author  = {Zitzler, Eckart and Deb, Kalyanmoy and Thiele, Lothar},
+ *   journal = {Evolutionary computation},
+ *   year    = {2000},
+ *   doi     = {10.1162/106365600568202}
+ * }
+ * @endcode
+ *
+ * @tparam arma::mat Type of matrix to optimize.
+ */
+  template<typename MatType = arma::mat>
+  class ZDT3
+  {
+    private:
+      size_t numObjectives;
+      size_t numVariables;
+
+    public:
+     //! Initialize the ZDT3
+    ZDT3() :
+        numObjectives(2),
+        numVariables(30),
+        objectiveF1(*this),
+        objectiveF2(*this)
+    {/* Nothing to do here. */}
+
+    /**
+     * Evaluate the objectives with the given coordinate.
+     *
+     * @param coords The function coordinates.
+     * @return arma::Col<typename MatType::elem_type>
+     */
+    arma::Col<typename MatType::elem_type> Evaluate(const MatType& coords)
+    {
+      double pi = arma::datum::pi;
```

Comment:
I would probably use it directly, since it's only used once and it's already an alias.